Test/45 tests edición de voto

manmunesp12 opened the pull request.

javivm17 committed:
Merge branch 'feature/8-votaciones_panel' into feature/29-depliegue_react_heroku

javivm17 committed:
feat: Añade configuración despliegue React

Se ha añadido al fichero Procfile la configuración necesaria para que se despliegue React

Co-Authored-By: Evelyn Gisele Yugsi Yugsi <61315703+eveyugyug@users.noreply.github.com>
Co-Authored-By: Javi Martínez <55922195+Javitoox@users.noreply.github.com>

Javitoox committed:
feat: Modifica la rama de despliegue

Se ha configurado github actions para que la rama actual pueda ser desplegada en heroku.

Co-Authored-By: Evelyn Gisele Yugsi Yugsi <61315703+eveyugyug@users.noreply.github.com>
Co-Authored-By: Javier Vilariño Mayo <44314228+javivm17@users.noreply.github.com>

Javitoox committed:
feat: Modifica ramas de despliegue

Se ha eliminado la condición que comprobaba la rama que podía ser desplegada por heroku para poder probarlo correctamente en este momento.

Co-Authored-By: Evelyn Gisele Yugsi Yugsi <61315703+eveyugyug@users.noreply.github.com>
Co-Authored-By: Javier Vilariño Mayo <44314228+javivm17@users.noreply.github.com>

javivm17 committed:
test: Añade configuración para testing del frontend

Se ha añadido la librería Cypress para realizar testing end-to-end y comprobar la funcionalidad del frontend. Se ha realizado un test para comprobar que la aplicación se encuentra disponible.

javivm17 committed:
test: Añade nuevos test de backups

Se han añadidos nuevos test end-to-end para comprobar la correcta funcionalidad de los backups.

javivm17 committed:
fix: Soluciona el problema de visualización de la API

Se ha solucionado el problema que no permitía la visualización de la API, esto era debido a que los nuevos end-points no tenían ningún serializer.

Commit:
test: Implementa test unitario edición de voto

Se ha implementado un test unitario que comprueba el correcto funcioanmiento
de la nueva funcionalidad de edición de voto, el cual se encarga de vota 2
dos veces y comprobar que se ha registrado correctamente la edición de voto.

Co-authored-by: Daniel Castroviejo Naranjo <dancasnar1@alum.us.es>

Javitoox committed:
fix: Añade correcciones

Se han implementado los cambios necesarios para que los tests se ejecuten correctamente.

El error ocurría ya que se tuvo que modificar las vistas de census para poder implementar la nueva funcionalidad del panel de control.

Para la corrección se ha modificado el test que daba error, así como el método de la vista de census que se modificó para mejorarlo y que no devuelva error cuando un usuario no exista.

Javitoox committed:
Merge branch 'fix/38-test_fallido_censo' into test/40-test_panel

Javitoox committed:
test: Añade tests

Se han implementado los tests referentes a las vistas, a la prueba de conexión, a las votaciones y a la página de inicio del panel de control que han sido considerados necesarios.

dancasnar1 committed:
test: Implementa nuevo tests

Se ha implementado un test que comprueba que la funcionalidad de edición de
voto no se permite realizarla en caso que la votación haya sido cerrada,
aunque se haya votado previamente antes de cerrarse.

Co-authored-by: Manuel Muñoz España <manmunesp1@alum.us.es>

Javitoox committed:
Merge branch 'test/40-test_panel' into feature/29-depliegue_heroku

Javitoox committed:
fix: Soluciona problema con las dependencias

Se ha modificado el fichero package.json para que el despliegue de la aplicación de React funcione correctamente.

Había surgido un problema con las dependecias de la librería de tests 'cypress'.

Co-Authored-By: Javier Vilariño Mayo <44314228+javivm17@users.noreply.github.com>

Javitoox committed:
Merge branch 'test/40-test_panel' into feature/29-depliegue_heroku

Javitoox committed:
chore: Añade cambios de configuración para heroku

Se han añadido cambios en la integración continua para el despliegue con heroku.

Co-Authored-By: Javier Vilariño Mayo <44314228+javivm17@users.noreply.github.com>

Javitoox committed:
chore: Añade cambios de configuración para heroku

Se han añadido cambios en la integración continua para el despliegue con heroku.

Co-Authored-By: Javier Vilariño Mayo <44314228+javivm17@users.noreply.github.com>

Javitoox committed:
chore: Añade cambios de configuración para heroku

Se han añadido cambios en la integración continua para el despliegue con heroku.

Co-Authored-By: Javier Vilariño Mayo <44314228+javivm17@users.noreply.github.com>

Javitoox committed:
chore: Añade cambios de configuración para heroku

Se han añadido cambios en la integración continua para el despliegue con heroku.

Co-Authored-By: Javier Vilariño Mayo <44314228+javivm17@users.noreply.github.com>

Javitoox committed:
chore: Añade cambios de configuración para heroku

Se han añadido cambios en la integración continua para el despliegue con heroku.

Co-Authored-By: Javier Vilariño Mayo <44314228+javivm17@users.noreply.github.com>

Javitoox committed:
chore: Añade cambios de configuración para heroku

Se han añadido cambios en la integración continua para el despliegue con heroku.

Co-Authored-By: Javier Vilariño Mayo <44314228+javivm17@users.noreply.github.com>

Javitoox committed:
chore: Añade cambios de configuración para heroku

Se han añadido cambios en la integración continua para el despliegue con heroku.

Co-Authored-By: Javier Vilariño Mayo <44314228+javivm17@users.noreply.github.com>

Javitoox committed:
chore: Añade cambios de configuración para heroku

Se han añadido cambios en la integración continua para el despliegue con heroku.

Co-Authored-By: Javier Vilariño Mayo <44314228+javivm17@users.noreply.github.com>

Javitoox committed:
chore: Añade cambios de configuración para heroku

Se han añadido cambios en la integración continua para el despliegue con heroku.

Co-Authored-By: Javier Vilariño Mayo <44314228+javivm17@users.noreply.github.com>

Javitoox committed:
chore: Añade cambios de configuración para heroku

Se han añadido cambios en la integración continua para el despliegue con heroku.

Co-Authored-By: Javier Vilariño Mayo <44314228+javivm17@users.noreply.github.com>

Javitoox committed:
chore: Añade cambios de configuración para heroku

Se han añadido cambios en la integración continua para el despliegue con heroku.

Co-Authored-By: Javier Vilariño Mayo <44314228+javivm17@users.noreply.github.com>

javivm17 committed:
chore: Añade cambios de configuración para heroku

Se han añadido cambios en la integración continua para el despliegue con heroku.

javivm17 committed:
chore: Añade cambios de configuración para heroku

Se han añadido cambios en la integración continua para el despliegue con heroku.

javivm17 committed:
chore: Añade cambios de configuración para heroku

Se han añadido cambios en la integración continua para el despliegue con heroku.

javivm17 committed:
chore: Añade cambios de configuración para heroku

Se han añadido cambios en la integración continua para el despliegue con heroku.

javivm17 committed:
chore: Añade configuracion local y remota de la API en el fronend

Se ha añadido la posibilidad de usar la API remota o la local.
Para lanzar el frontend usando la local seria de la siguiente forma:
- npm run start:local
Para lanzar el frontend usando la remota sería de la siguiente forma:
- npm start

javivm17 committed:
chore: Añade ejemplo de backup

Se ha añadido un ejemplo de backup para visualizarlo en el frontend ya que la base de datos remota no permite ni la creación ni la restauración. Además añade la inclusión de las variables de entorno

javivm17 committed:
chore: Añade cambios de configuración para heroku

Se han añadido cambios en la integración continua para el despliegue con heroku.

javivm17 committed:
chore: Añade cambios de configuración para heroku

Se han añadido cambios en la integración continua para el despliegue con heroku.

javivm17 committed:
chore: Añade cambios de configuración para heroku

Se han añadido cambios en la integración continua para el despliegue con heroku.

Javitoox committed:
chore: Borra backup

Se ha borrado un directorio de backups para evitar errores de directorios en local.

Javitoox committed:
chore: Borra Backup

Se ha borrado un directorio de backups para evitar errores de directorios en local.

Javitoox committed:
chore: Añade información

Se ha añadido información junto a la carpeta de backups, en la cual se encontrarán todos los backups generados por el sistema.

Javitoox committed:
Merge branch 'main' into feature/29-depliegue_heroku

Javitoox committed:
chore: Modifica backups

Se ha modificado la configuración de los backups para evitar conflictos y otras partes del código para el correcto funcionamiento del sistema.

NoSoyDani reviewed (approved):
Los test se ejecutan correctamente.

Javitoox committed:
chore: Añade cambios de configuración para heroku

Se han añadido cambios en la integración continua para el despliegue con heroku.

Javitoox committed:
chore: Modifica backup

Se ha modificado la configuración de los backups para evitar conflictos y otras partes del código para el correcto funcionamiento del sistema.

Javitoox committed:
chore: Modifica backups

Se ha modificado la configuración de los backups para evitar conflictos y otras partes del código para el correcto funcionamiento del sistema.

Javitoox committed:
chore: Modifica backups

Se ha modificado la configuración de los backups para evitar conflictos y otras partes del código para el correcto funcionamiento del sistema.

Javitoox committed:
chore: Modifica backups

Se ha modificado la configuración de los backups para evitar conflictos y otras partes del código para el correcto funcionamiento del sistema.

Javitoox committed:
chore: Modifica backups

Se ha modificado la configuración de los backups para evitar conflictos y otras partes del código para el correcto funcionamiento del sistema.

Javitoox committed:
chore: Añade cambios de configuración para heroku

Se han añadido cambios en la integración continua para el despliegue con heroku.

Javitoox committed:
chore: Añade cambios de configuración para heroku

Se han añadido cambios en la integración continua para el despliegue con heroku.

Javitoox committed:
chore: Añade cambios de configuración para heroku

Se han añadido cambios en la integración continua para el despliegue con heroku.

Javitoox committed:
chore: Añade cambios de configuración para heroku

Se han añadido cambios en la integración continua para el despliegue con heroku.

Javitoox committed:
chore: Añade cambios de configuración para heroku

Se han añadido cambios en la integración continua para el despliegue con heroku.

Concretamente se ha integrado en coverage de los tests de React.

Javitoox committed:
chore: Añade cambios de configuración para heroku

Se han añadido cambios en la integración continua para el despliegue con heroku.

Javitoox committed:
chore: Añade cambios de configuración para heroku

Se han añadido cambios en la integración continua para el despliegue con heroku.

Javitoox committed:
chore: Añade cambios de configuración para heroku

Se han añadido cambios en la integración continua para el despliegue con heroku.

Javitoox committed:
chore: Añade cambios de configuración para heroku

Se han añadido cambios en la integración continua para el despliegue con heroku.

Javitoox committed:
chore: Añade cambios de configuración para heroku

Se han añadido cambios en la integración continua para el despliegue con heroku.

Javitoox committed:
chore: Añade integración de releases

Se ha añadido la configuración necesaria de github actions para crear releases.

Javitoox committed:
chore: Añade integración de releases

Se ha añadido la configuración necesaria de github actions para crear releases.

Javitoox committed:
chore: Añade integración de releases

Se ha añadido la configuración necesaria de github actions para crear releases.

Javitoox committed:
chore: Añade integración de releases

Se ha añadido la configuración necesaria de github actions para crear releases.

Javitoox committed:
chore: Añade integración de releases

Se ha añadido la configuración necesaria de github actions para crear releases.

Javitoox committed:
chore: Añade integración de releases

Se ha añadido la configuración necesaria de github actions para crear releases.

Javitoox committed:
chore: Añade integración de releases

Se ha añadido la configuración necesaria de github actions para crear releases.

Javitoox committed:
chore: Añade integración de releases

Javitoox committed:
chore: Añade integración de releases

Javitoox committed:
chore: Añade integración de releases

Javitoox committed:
chore: Añade integración de releases

Javitoox committed:
chore: Añade integración de releases

Javitoox committed:
chore: Añade cambios en despliegue de heroku

Se ha modificado la rama en la cual se permite hacer el despliegue de heroku.

Javitoox committed:
chore: Actualiza README

Se ha actualizado la información de despliegue local del sistema.

dancasnar1 committed:
Merge branch 'feature/29-depliegue_heroku' into test/45-test_edicion_voto

dancasnar1 committed:
chore: Resube test de edición de votos

Se han resubido los tests correspondientes con la edición de votos una vez
se ha actualizado la rama con los cambios correspondientes.

Co-authored-by: Manuel Muñoz España <manmunesp1@alum.us.es>

eveyugyug reviewed (approved):
Se ha comprobado el correcto funcionamiento de todos los tests del módulo de almacenamiento.

dancasnar1 committed:
fix: Elimina la duplicación de test

Se han eliminado los test duplicados.

Co-authored-by: Manuel Muñoz España <manmunesp1@alum.us.es>
